File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        if(c <= 2){
            return true;
        }

        int times = (int) Math.ceil(Math.sqrt(c));
        Set<Integer> check = new HashSet<>();
        check.add(0);

        for(int i=times; i>0; i--){
            int a = i * i;
            check.add(a);
            if(check.contains(c-a)){
                System.out.println(check);
                return true;
            }
        }

        return false;
    }
}

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        """
        sqrt(c - a**2)
        """
        l, r = 0, int(sqrt(c))
        while l <= r:
            m = l * l + r * r
            if m > c:
                r -= 1
            elif m < c:
                l += 1
            else:
                return True
        
        return False